classes_hw.py

```diff
@@ -0,0 +1,168 @@
+# Выполнение всех реализованных методов по подсчёту метрик
+# Можно придумать дополнительные метрики и реализовать их (по 1 баллу за каждую)
+
+# In[225]:
+
+
+class Statistics_Fasta:
+
+    def __init__(self, path):
+        self.path = path
+        with open(self.path):
+            self.sequences = list(SeqIO.parse(self.path, "fasta"))
+
+    # Подсчёт числа последовательностей в фаста файле
+
+    def sequences_number(self):
+        return len(self.sequences)
+
+    # Построение гистограммы длин последовательностей
+    def sequences_length_distribution(self):
+        sequences_lengths = np.array([len(i.seq) for i in self.sequences])
+        figure(figsize=(30, 15))
+        names, counts = np.unique(sequences_lengths, return_counts=True)
+        plt.bar(range(len(names)), counts)
+        plt.xticks(range(len(names)), names)
+        plt.title("Sequence length distribution")
+        plt.show()
+
+    # Подсчёт GC состава
+    def count_GC(self):
+        gcs = np.array([GC(j.seq) for j in self.sequences])
+        return gcs.mean()
+
+    # Построение гистограммы GC состава
+    def sequences_GC_distribution(self):
+        gcs = np.array([GC(j.seq) for j in self.sequences])
+        figure(figsize=(30, 15))
+        names, counts = np.unique(gcs, return_counts=True)
+        plt.bar(range(len(names)), counts)
+        plt.xticks(range(len(names)), names)
+        plt.title("GC content distribution")
+        plt.show()
+
+    # Построение гистограммы частоты 4-меров (по оси x каждый из возможных 4-меров, а по y - их частота)
+    def counts_4_mers(self):
+        possible_4_mers = dict()
+        k = 4
+        sequences = [s.seq for s in self.sequences]
+        for seq in sequences:
+            for i in range(len(seq) - k + 1):
+                kmr = seq[i:i + k]
+                if kmr in possible_4_mers:
+                    possible_4_mers[kmr] += 1
+                else:
+                    possible_4_mers[kmr] = 1
+        new = {k: v for k, v in sorted(possible_4_mers.items(), key=lambda x: x[1], reverse=True)}
+        y = list(new.values())
+        names = list(new.keys())
+        figure(figsize=(40, 10))
+        plt.bar(range(len(y)), y)
+        plt.xticks(range(len(names)), names, rotation='vertical')
+        plt.title("Distribution of 4-mers counts")
+        plt.show()
+
+    # Переопределение метода для вывода информации при принте (достаточно текста с указанием путя к файлу)
+    def __repr__(self):
+        return self.path
+
+    # Выполнение всех реализованных методов по подсчёту метрик
+    def runall(self):
+        if __name__ == '__main__':
+            Thread(target=self.sequences_number).start()
+            Thread(target=self.sequences_length_distribution).start()
+            Thread(target=self.count_GC).start()
+            Thread(target=self.counts_4_mers).start()
+            Thread(target=self.sequences_GC_distribution).start()
```

Comment:
Обычно эту конструкцию не пишут внутри функций или классов. Я бы написал метод `run_all` без этого, а потом просто бы вызвал его в глобальной области видимости под `if __name__ == "__main__:"`
```suggestion
    def runall(self):
            self.sequences_number()
            self.sequences_length_distribution()
            self.count_GC()
            self.counts_4_mers()
            self.sequences_GC_distribution()

if __name__ == "__main__":
    Statistics_fasta("sequences.fasta").run_all()
```
Потоки в питоне не работают параллельно, а скорее даже дольше, чем те же функции выполненые последовательно, поэтому их использование здесь абсолютно неуместно.